### PR TITLE
feat(new-primary-school): Implement no children page

### PR DESCRIPTION
**Pull request by birkirkristmunds**
[TS-816](https://dit-iceland.atlassian.net/browse/TS-816)

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/new-primary-school/src/lib/NewPrimarySchoolTemplate.ts [hunk 1]
```diff
@@ -73,13 +77,37 @@ const NewPrimarySchoolTemplate: ApplicationTemplate<
...
+              actions: 'setApproveExternalData',
+            },
+          ],
         },
       },
       [States.DRAFT]: {
         meta: {
           name: States.DRAFT,
           status: 'draft',
+
```
Comment (veronikasif) [inline libs/application/templates/new-primary-school/src/lib/NewPrimarySchoolTemplate.ts:110]: Remove new line
```suggestion
```

File: libs/application/templates/new-primary-school/src/lib/NewPrimarySchoolTemplate.ts [hunk 2]
```diff
@@ -73,13 +77,37 @@ const NewPrimarySchoolTemplate: ApplicationTemplate<
...
+                // No child found
+                if (!children || children.length === 0) {
+                  return false
+                }
+
+                // Check if the applicant has some children at primary school age
+                return children.some((child) =>
+                  isChildAtPrimarySchoolAge(child.nationalId),
+                )
+              },
```
Comment (veronikasif) [inline libs/application/templates/new-primary-school/src/lib/NewPrimarySchoolTemplate.ts:98]: Move functionality into `NewPrimarySchoolTemplateUtils.ts`?
```suggestion
              cond: (application) => canApply(),
```

File: libs/application/templates/new-primary-school/src/forms/Prerequisites.ts [hunk 3]
```diff
@@ -82,6 +84,25 @@ export const Prerequisites: Form = buildForm({
...
+              title: '',
+              description: newPrimarySchoolMessages.pre.noChildrenFoundReasons,
+            }),
+
+            // Empty submit field to hide all buttons in the footer
+            buildSubmitField({
+              id: '',
+              title: '',
+              actions: [],
+            }),
```
Comment (veronikasif) [inline libs/application/templates/new-primary-school/src/forms/Prerequisites.ts:103]: Is this still needed if you have `renderLastScreenButton` and `renderLastScreenBackButton`?
```suggestion
```

File: libs/application/templates/new-primary-school/src/lib/NewPrimarySchoolTemplate.ts [hunk 4]
```diff
@@ -73,13 +77,37 @@ const NewPrimarySchoolTemplate: ApplicationTemplate<
...
-          SUBMIT: States.DRAFT,
+          SUBMIT: [
+            {
+              target: States.DRAFT,
+              cond: (application) => {
+                const { children } = getApplicationExternalData(
+                  application?.application?.externalData,
+                )
+
+                //
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/lib/NewPrimarySchoolTemplate.ts]: Fixed

File: libs/application/templates/new-primary-school/src/lib/NewPrimarySchoolTemplate.ts (hunk 1 above)
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/lib/NewPrimarySchoolTemplate.ts]: Fixed

File: libs/application/templates/new-primary-school/src/lib/NewPrimarySchoolTemplate.ts (hunk 2 above)
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/lib/NewPrimarySchoolTemplate.ts]: Fixed

File: libs/application/templates/new-primary-school/src/forms/Prerequisites.ts (hunk 3 above)
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/forms/Prerequisites.ts]: Fixed

**Review by veronikasif — APPROVED** [review]
LGTM

**Pull request merged**